cns/restserver/api_test.go

```diff
@@ -246,8 +246,90 @@ func TestSetOrchestratorType_NCsPresent(t *testing.T) {
 					ContainerStatus: map[string]containerstatus{
 						"nc1": {},
 					},
-					ContainerIDByOrchestratorContext: map[string]string{
-						"nc1": "present",
+					ContainerIDByOrchestratorContext: map[string]Set{
+						"nc1": {"present": {}},
+					},
+				},
+			},
+			writer: httptest.NewRecorder(),
+			request: FirstRequest(http.NewRequestWithContext(
+				context.TODO(), http.MethodPost, cns.SetOrchestratorType, bytes.NewReader(
+					FirstByte(json.Marshal( //nolint:errchkjson //inline map, only using returned bytes
+						cns.SetOrchestratorTypeRequest{
+							OrchestratorType: "Kubernetes",
+							DncPartitionKey:  "partition1",
+							NodeID:           "node2",
+						}))))),
+			response: cns.Response{
+				ReturnCode: types.InvalidRequest,
+				Message:    "Invalid request since this node has already been registered as node1",
+			},
+			wanthttperror: false,
+		},
+		{
+			name: "Node already set, with no NCs, so registration should succeed",
+			service: &HTTPRestService{
+				state: &httpRestServiceState{
+					NodeID: "node1",
+				},
+			},
+			writer: httptest.NewRecorder(),
+			request: FirstRequest(http.NewRequestWithContext(
+				context.TODO(), http.MethodPost, cns.SetOrchestratorType, bytes.NewReader(
+					FirstByte(json.Marshal( //nolint:errchkjson //inline map, only using returned bytes
+						cns.SetOrchestratorTypeRequest{
+							OrchestratorType: "Kubernetes",
+							DncPartitionKey:  "partition1",
+							NodeID:           "node2",
+						}))))),
+			response: cns.Response{
+				ReturnCode: types.Success,
+				Message:    "",
+			},
+			wanthttperror: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var resp cns.Response
+			// Since this is global, we have to replace the state
+			oldstate := svc.state
+			svc.state = tt.service.state
+			mux.ServeHTTP(tt.writer, tt.request)
+			// Replace back old state
+			svc.state = oldstate
+
+			err := decodeResponse(tt.writer, &resp)
+			if tt.wanthttperror {
+				assert.NotNil(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.response, resp)
+		})
+	}
+}
+
+func TestSetOrchestratorType_DualNCsPresent(t *testing.T) {
```

Comment:
we agreed last time that we don't need set orchestrator type test for dual NC. Remove this newly added test.
where are the tests for 
TestCreateNetworkContainerDualNic and
TestGetAllNetworkContainerByOrchestratorContext?

Reply:
I think TestGetAllNetworkContainers == TestGetAllNetworkContainerByOrchestratorContext() test case